common/src/java/org/apache/hive/http/HttpServer.java

```diff
@@ -534,9 +542,110 @@ private void setupCORSFilter(Builder b) {
   }
 
   /**
-   * Create a channel connector for "http/https" requests
+   * Creates a port connector and initializes a web application that processes requests through the newly 
+   * created port connector.
+   * 
+   * @param builder - The builder object used to configure and create the port connector and web application.
+   * @return ContextHandlerCollection - A collection of request handlers associated with the new port connector,
+   *         which includes the newly initialized web application.
+   */
+  public ContextHandlerCollection addWebApp(Builder builder) throws IOException {
+    WebAppContext webAppContext = createWebAppContext(builder);
+    initWebAppContext(builder, webAppContext);
+    RewriteHandler rwHandler = createRewriteHandler(builder, webAppContext);
+    
+    ContextHandlerCollection portHandler = new ContextHandlerCollection();
+    ServerConnector connector = addChannelConnector(threadPool.getQueueSize(), builder);
+    portHandler.addHandler(rwHandler);
+
+    for (Pair<String, Class<? extends HttpServlet>> p : builder.servlets) {
+      addServlet(p.getFirst(), "/" + p.getFirst(), p.getSecond(), webAppContext);
```

Comment:
could we use `getKey/Value` Pair method signature?

Reply:
That was in the original code, fixed.